micropython/drivers/led/neopixel/neopixel.py

```diff
@@ -8,10 +8,16 @@ class NeoPixel:
     # G R B W
     ORDER = (1, 0, 2, 3)
 
-    def __init__(self, pin, n, bpp=3, timing=1):
+    def _clamp(self, v, c_min, c_max):
+        return min(max(v, c_min), c_max)
+
+    def __init__(self, pin, n, bpp=3, timing=1, brightness=None):
         self.pin = pin
         self.n = n
         self.bpp = bpp
+        self.brightness = None
```

Comment:
We should rename this to `self.b` -- class members end up allocating qstrs, so using small names is better.

It's also smaller to write

```
self.b = None if brightness is None else min(max(brightness), 0, 1)
```

rather than

```
self.b = None
if brightness is not None:
   self.b = min(max(brightness), 0, 1)
```